clippy_lints/src/write.rs

```diff
@@ -187,10 +192,35 @@ declare_lint_pass!(Write => [
 ]);
 
 impl EarlyLintPass for Write {
+    fn check_item(&mut self, _: &EarlyContext<'_>, item: &Item) {
+        if let ItemKind::Impl {
+            of_trait: Some(trait_ref),
+            ..
+        } = &item.kind
+        {
+            let trait_name = trait_ref
+                .path
+                .segments
+                .iter()
+                .last()
+                .expect("path has at least one segment")
+                .ident
+                .name
+                .as_str();
+            if trait_name == "Debug" {
```

Comment:
```suggestion
            if trait_name == sym!(Debug) {
```

Also remove `as_str()` above.